### ISSUE TITLE
Passing C# field through javascript and back into C# results in null

### ISSUE THREAD

**Issue by SnowSnakz** [JavascriptNet/Javascript.Net#87]
Here is my setup:

I plan to use JavaScript as the language for modding my game, I currently have it setup to create a javascript context with 3 variables: `log`, `mod` and `require`, `require` isn't being used yet...

`mod` is a C# object with the field I am trying to use.
`log` is also a C# object with functions that redirect to log4net, (mainly `info(string format, object[] objs)`)

Here is the issue, in my javascript file I have:
```
log.info("Hello from {0}!", [mod.display_name]);
```
Which results in:
```
Hello from !
```
Even though `mod`'s C# object has the field `display_name` set to `"sandbox"`
I even checked with the debugger, the C# instance of `mod` is correct, I debugged the C# function for `log.info` and noticed that mod.display_name was evaluating to null before being passed through the function. Why is this happening? And how can I fix it?

**Comment by oliverbock** [comment]
I think only `public` properties will go through to JavaScript.  Obviously if your `displayr_name` is null then that is your problem.

**Comment by SnowSnakz** [comment]
> I think only `public` properties will go through to JavaScript. Obviously if your `displayr_name` is null then that is your problem.

The field is both public and not null.

**Comment by oliverbock** [comment]
Try to reduce this to a simple example that you can post here.

**Comment by SnowSnakz** [comment]
I can't even seem to setup the project anymore, Visual Studio just crashes when I load my pre-existing project and when I install the NuGet package, the Noesis.Javascript namespaces don't exist.... This has to be a bug with my Visual Studio installation.

**Comment by SnowSnakz** [comment]
Reguardless, I was using *fields*, and I just now noticed that you said "properties"... So if I am not mistaken by "property" you mean specifically this:
```
public Type PropertyName { get; set; }
```

And not this:
```
public Type FieldName;
```

**Comment by oliverbock** [comment]
Yes, I don't think fields are supported.  You get null rather than an error when a property does not exist.  (This is convenient sometimes, and not other times.  Your object can also store arbitrary new properties, if I remember correctly.)
